hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/ColumnStatIndexTestBase.scala

```diff
@@ -156,12 +167,61 @@ class ColumnStatIndexTestBase extends HoodieSparkClientTestBase {
       }).asJava,
       indexSchema
     )
+
+    if (metaClient.getTableConfig.getTableType == HoodieTableType.COPY_ON_WRITE) {
+      baseFilesDf // COW table
+    } else {
+      val allLogFiles = filegroupList.stream().flatMap(fileGroup => fileGroup.getAllFileSlices)
+        .flatMap(fileSlice => fileSlice.getLogFiles)
+        .collect(Collectors.toList[HoodieLogFile])
+      if (allLogFiles.isEmpty) {
+        baseFilesDf // MOR table, but no log files.
+      } else {
+        val colsToGenerateStats = indexedCols // check for included cols
+        val writerSchemaOpt = LogFileColStatsTestUtil.getSchemaForTable(metaClient)
+        val latestCompletedCommit = metaClient.getActiveTimeline.getCommitsTimeline.filterCompletedInstants().lastInstant().get().getTimestamp
+        baseFilesDf.union(getColStatsForLogFiles(allLogFiles, latestCompletedCommit,
+          scala.collection.JavaConverters.seqAsJavaList(colsToGenerateStats),
+          metaClient,
+          writerSchemaOpt: org.apache.hudi.common.util.Option[Schema],
+          HoodieMetadataConfig.MAX_READER_BUFFER_SIZE_PROP.defaultValue(),
+          indexSchema))
+      }
+    }
+  }
+
+  protected def getColStatsForLogFiles(logFiles: List[HoodieLogFile], latestCommit: String, columnsToIndex: util.List[String],
```

Comment:
Use `For` or `From`, not both, to be consistent 
```suggestion
  protected def getColStatsForLogFiles(logFiles: List[HoodieLogFile], latestCommit: String, columnsToIndex: util.List[String],
```